src/status_im2/contexts/shell/events.cljs

```diff
@@ -127,3 +138,87 @@
   {:events [:change-shell-nav-bar-color]}
   [_ color]
   {:merge-options {:id "shell-stack" :options {:navigationBar {:backgroundColor color}}}})
+
+(rf/defn shell-navigate-to
+  {:events [:shell/navigate-to]}
+  [{:keys [db]} go-to-view-id screen-params animation hidden-screen?]
+  (if (shell.utils/shell-navigation? go-to-view-id)
+    (let [current-view-id (:view-id db)
+          community-id    (get-in db [:chats screen-params :community-id])]
+      {:db         (assoc-in
+                    db
+                    [:shell/floating-screens go-to-view-id]
+                    {:id             screen-params
+                     :community-id   community-id
+                     :hidden-screen? hidden-screen?
+                     :animation      (or animation
+                                         (case current-view-id
+                                           :shell shell.constants/open-screen-with-shell-animation
+                                           :chat  shell.constants/open-screen-without-animation
+                                           shell.constants/open-screen-with-slide-animation))})
+       :dispatch-n (cond-> []
+                     (not hidden-screen?)
+                     (conj [:set-view-id go-to-view-id])
+                     (and (= go-to-view-id shell.constants/community-screen)
+                          (not hidden-screen?)
+                          (:current-chat-id db))
+                     (conj [:chat/close]))})
+    {:db          (assoc db :view-id go-to-view-id)
+     :navigate-to go-to-view-id}))
+
+(rf/defn shell-navigate-back
+  {:events [:shell/navigate-back]}
+  [{:keys [db]}]
+  (let [chat-screen-open?      (shell.utils/floating-screen-open? shell.constants/chat-screen)
+        community-screen-open? (shell.utils/floating-screen-open? shell.constants/community-screen)
+        current-chat-id        (:current-chat-id db)
+        community-id           (when current-chat-id
+                                 (get-in db [:chats current-chat-id :community-id]))]
+    (if (and (not @navigation.state/curr-modal)
```

Comment:
Imo it's best to address issues like this before they get merged in. It's likely it never gets improved otherwise.

Reply:
> Imo it's best to address issues like this before they get merged in. It's likely it never gets improved otherwise.

Hi @ilmotta mentions issue in our navigation state management and migrating to better method using app db (issue is not particular related to PR)

Reply:
I mean, it's related to this PR since it's code being merged in `status_im2`, which is a directory we want to write code using superior standards than the old namespaces.

But as I shared previously, I'm okay if this gets merged, maybe others feel stronger about it. Let's just be aware that mutable re-frame event handlers is something we should basically never need to do.